### PR TITLE
Updated notebook, and nemo2 checkpoint with geneformer

### PR DESCRIPTION
**Pull request by jstjohn**
## Changes
- Added new checkpoint for the 10m geneformer model
- Updated the load_context call to only grab the model config and bypass issues created when the entire trainer is re-instantiated.

### CODE REVIEW

File: sub-packages/bionemo-llm/src/bionemo/llm/model/config.py [hunk 1]
```diff
@@ -86,8 +85,9 @@ def load_settings_from_checkpoint(self, initial_ckpt_path: str) -> None:
         logger.warn(f"Loading {self.initial_ckpt_path}")
         # 1. get the config
         # TODO type(self) is probably not correct, maybe make the class name of the config to load an argument?
```
Comment (skothenhill-nv) [inline sub-packages/bionemo-llm/src/bionemo/llm/model/config.py:87]: can we cleanup this comment now?

**Review by polinabinder1 — APPROVED** [review]

File: sub-packages/bionemo-llm/src/bionemo/llm/model/config.py [hunk 2]
```diff
@@ -86,8 +85,9 @@ def load_settings_from_checkpoint(self, initial_ckpt_path: str) -> None:
...
-        cfg_trainer_ctx: TrainerContext = io.load_context(Path(initial_ckpt_path) / "context")
-        initial_config: MegatronBioNeMoTrainableModelConfig = cfg_trainer_ctx.model.config
+        initial_config: MegatronBioNeMoTrainableModelConfig = io.load_context(
+            path=Path(initial_ckpt_path) / "context", subpath="model.config"
+        )  # type: ignore
```
Comment (skothenhill-nv) [inline sub-packages/bionemo-llm/src/bionemo/llm/model/config.py:90]: so `subpath` is how we pickout specific things from the checkpoint?? nice!

**Review by skothenhill-nv — APPROVED** [review]

**Review by pstjohn — APPROVED** [review]

File: sub-packages/bionemo-core/src/bionemo/core/data/resources/geneformer.yaml [hunk 3]
```diff
@@ -20,6 +20,18 @@
     final shape of 25472 for GPU efficiency. See https://wandb.ai/clara-discovery/scFM_v9/runs/4ij9ghox and converted
     to nemo2 format with https://github.com/NVIDIA/bionemo-framework/pull/180
 
+- tag: 10M_241113:2.0
+  ngc: null
```
Comment (jstjohn) [inline sub-packages/bionemo-core/src/bionemo/core/data/resources/geneformer.yaml:24]: `nvstaging/clara/geneformer_10m:2.1` see https://nvidia.slack.com/archives/C074Z808N05/p1731594944514839?thread_ts=1731525485.146139&cid=C074Z808N05

File: sub-packages/bionemo-geneformer/tests/bionemo/geneformer/test_stop_and_go.py [hunk 4]
```diff
@@ -97,7 +97,7 @@ class TestGeneformerStopAndGo(stop_and_go.StopAndGoHarness):
     limit_val_batches: int = 2
     lr: float = 1e-4
     precision: Literal["16-mixed", "bf16-mixed", "32"] = MODEL_PRECISION
-    train_val_output_atol: float = 2e-2
+    train_val_output_atol: float = 2.5e-2
```
Comment (jstjohn) [inline sub-packages/bionemo-geneformer/tests/bionemo/geneformer/test_stop_and_go.py:100]: Wait for @pstjohn to merge his PR which does this as well.

**Pull request merged**